content/articles/create-a-secure-home-connection-using-pi-hole-and-docker/index.md

```diff
@@ -0,0 +1,153 @@
+
+2. After that, you'll be asked which external DNS server you want to use. Here, Cloudflare and Google are both good free solutions. Using your arrow keys, select the provider you want to use, then confirm using the enter key.
+
+3. Your choice of adblocking lists will be asked of you in the next round. Pi-hole comes with four default lists, which you should leave chosen, but you can enable or disable any of them by selecting them and pressing space on your keyboard.
+To continue, press the tab key to select OK, then enter.
+
+4. Pi-hole will prevent advertisements on IPv4 and IPv6 connections by default. If you don't want to alter anything, leave the default options selected, press the tab to select OK, and then press enter.
+
+5. The following step will require you to confirm that the IP address and IP gateway (most likely your local router) displayed are correct for the Pi-static hole's IP setting. To update your settings, press enters on Yes if this is right, or No if it isn't.
+
+6. Pi-hole will alert you about any possible IP conflicts. Accept the warning and continue by pressing the enter key.
+
+7. Pi-hole comes with a web browser admin panel that you may use to configure and monitor it by default. This is the recommended option for installation, and it is the default choice for the installation process. To continue, press the tab and then enter OK. This will also be required at a later stage to validate the installation of a web server for the admin interface.
+
+8. If you want to log inquiries, Pi-hole will ask you. This is useful since you'll be able to monitor which domains Pi-hole is blocking and how frequently they're banned. This is already selected, so press tab and enter to confirm.
+
+9. You can choose how detailed your Pi-hole statistics should be. In the documentation for Pi-hole, there are five levels, each of which is discussed in great detail. To ensure that you're happy with the default option of showing everything, press the tab and enter.
+
+10. The Pi-hole installation will begin once you've chosen your preferred logging level. Once the installation is complete, the terminal will show a final confirmation message. This message will include information about logging in to the web portal and a password that will be generated automatically for you. Enter after pressing the tab. 
+
+#### Installing Pi-hole as a Docker container 
+If you want, instead of installing Pi-hole using the script above, you can use Docker to run it in a separate Docker software container. Your Raspberry Pi must be installed using Docker first.
+
+#### Installation of a docker container
+1. Open a terminal window and type the following instructions to install Docker on your Raspberry Pi:
+
+```bash
+curl -fsSL https://get.docker.com -o get-docker.sh
+sh get-docker.sh

```

Comment:
You previously mentioned that it might be dangerous to run a script directly from the internet. As such, provide the alternative commands for manual installation of a docker container as well.